react-backend/client/src/components/authentication/logincb.js

```diff
@@ -51,14 +51,15 @@ class CBlogin extends Component {
             field blank before continuing.
           </span>
         );
-        if (data === 'true') {
-          console.log("Success, now it's time to use cookies");
+        if (data.success === true) {
+          localStorage.setItem('token', data.token);
+          this.props.setLoggedIn();
           this.props.history.push('/visitor');
-        } else if (data === 'email') {
+        } else if (data.reason === 'email') {
           this.setError([EMAIL_ERROR]);
-        } else if (data === 'noinput') {
+        } else if (data.reason === 'noinput') {
           this.setError([NO_INPUT_ERROR]);
-        } else if (data === 'false') {
+        } else if (data.success === false) {
```

Comment:
this can just be an else statement. 

The only way the code would ever get to this condition is if `data.success` isn't true (since it's always a boolean, that means it must be false), and if `data.reason` is not 'email' or 'noinput', so you don't have to check that `data.success === false` here, as if it gets to this bit of code, it always will be false.